storage/diemdb/src/transaction_store/mod.rs

```diff
@@ -46,6 +50,18 @@ impl TransactionStore {
         Ok(None)
     }
 
+    /// Gets the version of a transaction by its hash.
+    pub fn get_transaction_version_by_hash(
+        &self,
+        hash: &HashValue,
+        ledger_version: Version,
+    ) -> Result<Option<Version>> {
+        Ok(match self.db.get::<TransactionByHashSchema>(hash)? {
+            Some(version) if version <= ledger_version => Some(version),
```

Comment:
I think this if condition is really confusing, we've found txn version, why I will pass in a ledger_version to filter it out.